### DISCUSSION TITLE
2023-07-13: React functionality

### DISCUSSION THREAD

**Discussion by Fade253** [coreybutler/node-windows#355] [General] [answered]
**Is your feature request related to a problem? Please describe.**
i am trying to run node-windows with react but have no idea which file to run it with, i have tried the development server but that didnt have any luck at all, i have tried index, it hates imports, and tried the setup to the C# side i have that isnt the main backend and nothing works

**Describe the solution you'd like**
Allow for just setting it up with index and not having to do anything else for react

**Describe alternatives you've considered**

**Additional context**
if you cant make it work for index at least fix trying to run it with the development server as it gets an internal error trying to find the script folder the server is IN

<table>
  <tr>
    <td>Please use the reactions to cast a vote in favor of or against this feature suggestion &raquo;</td>
    <td><img src="https://github.com/metadoc/metadoc.io/raw/main/.github/ISSUE_TEMPLATE/vote.png" width="75"/></td>
  </tr>
</table>

**Comment by coreybutler** [comment, marked as answer]
I'm not entirely clear what you're trying to do here, so please clarify if I'm off base in my response.

node-windows creates and manages background services, such as web servers and APIs. React is a frontend framework that runs in browsers, not in Node. React has some node-based build tools (i.e. `create-react-app`), but those aren't designed to be background services either.

node-windows would be more appropriate for a "main backend", i.e. the web server. It sounds like your backend is written in C#, so node-windows likely isn't the right fit for this scenario. If you had an Express/Restify/Fastify backend server, then node-windows would be a more appropriate fit.

If you need:

- A static web server to _develop_ frontends: [Fenix Web Server](https://preview.fenixwebserver.com)
- A static web server to serve _production_ React code: [Caddy Server](https://caddyserver.com/), NGINX, Apache, IIS, etc.
- Roll your own Node _server_: Node server + node-windows

I'm going to close this since I don't think it is in the scope of what node-windows is for, but I will convert this issue to a discussion.

**Reply by Fade253** [reply]
i was trying to run the development server as a service when i needed to run a server file as the service for it to run, sorry for taking up your time. works great when using it on the build version

**Reply by coreybutler** [reply]
No worries, glad you got things straightened out.

**Reply by MarkAurit** [reply]
well, you have a solution, but you very well don't.  trying to get the front end to play well with the back end often involves taking advantage of what others might see as a bug in the base system that will be "fixed" in a later version.  with these two systems being as popular as they are one would expect to see your situation addressed thoroughly on the various forums - and if they aren't that should be telling you something.
